skynet/modules/ttt/rag/vector_store.py

```diff
@@ -0,0 +1,122 @@
+import asyncio
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from langchain_core.documents import Document
+from langchain_core.vectorstores.base import VectorStore
+from langchain_huggingface import HuggingFaceEmbeddings
+
+from skynet.env import embeddings_model_path
+from skynet.logs import get_logger
+from skynet.modules.ttt.assistant.v1.models import RagConfig, RagPayload, RagStatus
+from skynet.modules.ttt.rag.constants import ERROR_RAG_KEY, RUNNING_RAG_KEY, STORED_RAG_KEY
+from skynet.modules.ttt.rag.web_crawler.main import crawl
+
+from ..persistence import db
+
+log = get_logger(__name__)
+
+
+class SkynetVectorStore(ABC):
+    embedding = HuggingFaceEmbeddings(
+        model_name=embeddings_model_path, model_kwargs={'device': 'cpu', 'trust_remote_code': True}
```

Comment:
Why was trust_remote_code needed?

Reply:
"The trust_remote_code=True flag is used for models whose code lives on the hub rather than natively in the Transformers library."